src/Compilers/CSharp/Portable/Binder/SwitchBinder.cs

```diff
@@ -560,7 +561,24 @@ private BoundSwitchSection BindSwitchSection(SwitchSectionSyntax node, Binder or
             var boundStatementsBuilder = ArrayBuilder<BoundStatement>.GetInstance();
             foreach (var statement in node.Statements)
             {
-                boundStatementsBuilder.Add(sectionBinder.BindStatement(statement, diagnostics));
+                var boundStatement = sectionBinder.BindStatement(statement, diagnostics);
+                if (boundStatement is BoundLocalDeclaration boundLocal)
```

Comment:
>if [](start = 16, length = 2)

Nit: consider extracting the checks to a helper method to keep the main flow clean.